src/pages/TradeV3/perps/DepositWithdrawNew.tsx

```diff
@@ -0,0 +1,345 @@
+        else if (index === 1) result = (+tokenAmount.uiAmountString * 50) / 100
+        else if (index === 2) result = (+tokenAmount.uiAmountString * 75) / 100
+        else result = +tokenAmount.uiAmountString
+        setAmount(String(result))
+      } else {
+        const avail = traderInfo.maxWithdrawable
+        if (!avail || Number.isNaN(+avail)) {
+          setAmount('0.00')
+          return
+        }
+        let result = 0
+        if (index === 0) result = (+avail * 25) / 100
+        else if (index === 1) result = (+avail * 50) / 100
+        else if (index === 2) result = (+avail * 75) / 100
+        else result = +avail
+        setAmount((Math.floor(result * 1000) / 1000).toString())
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (isDevnet) setAmount('500')
+  }, [isDevnet])
+
+  const handleInputChange = (e) => {
+    if (isDevnet) {
+      setAmount('500')
+    } else {
+      const t = e.target.value
+      if (!isNaN(+t)) {
+        e.target.value = t.indexOf('.') >= 0 ? t.substr(0, t.indexOf('.')) + t.substr(t.indexOf('.'), 5) : t
+        setAmount(e.target.value)
+      }
+    }
+  }
+  const handleSubmit = async () => {
+    try {
+      const answer = convertToFractional(amount)
+      setIsLoading(true)
+      const response = tradeType === 'deposit' ? await depositFunds(answer) : await withdrawFunds(answer)
+      if (response === null) {
+        setIsLoading(false)
+        return
+      }
+
+      // this will update balances
+      fetchAccounts()
+
+      if (response && response.txid) {
+        setIsLoading(false)
+        setDepositWithdrawModal(false)
+      }
+    } catch (e) {
+      console.log(e)
+      setIsLoading(false)
+    }
+  }
+  const checkDisabled = () => {
+    if (!publicKey) return true
+    if (isDevnet) {
+      if (!traderInfo.traderRiskGroup) return false
+      if (traderInfo.traderRiskGroup.totalDeposited.toJSON().m !== '0') return true
+      return false
+    }
+    if (tradeType !== 'deposit') {
+      if (!traderInfo.marginAvailable || +traderInfo.marginAvailable < +amount || !amount || !+amount) return true
+    } else {
+      if (
+        !tokenAmount ||
+        isLoading ||
+        !tokenAmount.uiAmountString ||
+        +tokenAmount.uiAmountString < +amount ||
+        !amount ||
+        !+amount
+      )
+        return true
+    }
+  }
+
+  const menus = (
+    <div>
+      {perpTokenList.map((item, index) => (
+        <DropdownMenuItem onClick={() => setPerpToken(item)} key={index}>
+          <p className={cn('font-bold w-[450px] cursor-pointer')}>
+            <Tokens {...item} />
+          </p>
+        </DropdownMenuItem>
+
+        // <Menu.Item
+        //   key={index}
+        //   onClick={() => {
+        //     setPerpToken(item)
+        //   }}
+        // >
+        //   <Tokens {...item} />
+        // </Menu.Item>
+      ))}
+    </div>
+  )
+  const args = {
+    rightItem: (
```

Comment:
would a better flow be to rather pass a child to input group and the group does the ordering? or how do you think (comp lib question)